text/2018-10-25-tikv-client-rust.md

```diff
@@ -0,0 +1,389 @@
+
+To use the Transactional Key-Value API, take the following steps:
+
+1. Create an instance of Config to specify endpoints of PD (Placement Driver) and optional security config
+
+    ```rust
+    let config = Config::new(vec!["127.0.0.1:2379"]);
+    ```
+
+2. Create a Transactional Key-Value client.
+
+    ```rust
+    let client = TxnClient::new(&config);
+    ```
+
+3. (Optional) Modify data using a Transaction.
+
+    The lifecycle of a Transaction is: _begin → {get, set, delete, scan} → {commit, rollback}_.
+
+4. Call the Transactional Key-Value API's methods to access the data on TiKV. The Transactional Key-Value API contains the following methods:
+
+    ```rust
+    fn begin(&self) -> KvFuture<Transaction>;
+
+    fn get<K>(&self, key: K) -> KvFuture<Value>
+    where
+        K: AsRef<Key>;
+
+    fn set<P>(&mut self, pair: P) -> KvFuture<()>
+    where
+        P: Into<KvPair>;
+
+    fn delete<K>(&mut self, key: K) -> KvFuture<()>
+    where
+        K: AsRef<Key>;
+
+    fn seek<K>(&self, key: K) -> KvFuture<Scanner>
+    where
+        K: AsRef<Key>;
+
+    fn commit(self) -> KvFuture<()>;
+
+    fn rollback(self) -> KvFuture<()>;
+    ```
+
+### Usage example of the Transactional Key-Value API
+
+    ```rust
+    extern crate futures;
+    extern crate tikv_client;
+
+    use futures::{Async, Future, Stream};
+    use tikv_client::transaction::{Client, Mutator, Retriever, TxnClient};
+    use tikv_client::*;
+
+    fn puts<P, I>(client: &TxnClient, pairs: P)
+    where
+        P: IntoIterator<Item = I>,
+        I: Into<KvPair>,
+    {
+        let mut txn = client.begin().wait().expect("Could not begin transaction");
+        let _: Vec<()> = pairs
+            .into_iter()
+            .map(Into::into)
+            .map(|p| {
+                txn.set(p).wait().expect("Could not set key value pair");
+            }).collect();
+        txn.commit().wait().expect("Could not commit transaction");
+    }
+
+    fn get(client: &TxnClient, key: &Key) -> Value {
+        let txn = client.begin().wait().expect("Could not begin transaction");
+        txn.get(key).wait().expect("Could not get value")
+    }
+
+    fn scan(client: &TxnClient, start: &Key, limit: usize) {
+        let txn = client.begin().wait().expect("Could not begin transaction");
+        let mut scanner = txn.seek(start).wait().expect("Could not seek to start key");
+        let mut limit = limit;
+        loop {
+            if limit == 0 {
+                break;
+            }
+            match scanner.poll() {
+                Ok(Async::Ready(None)) => return,
+                Ok(Async::Ready(Some(pair))) => {
+                    limit -= 1;
+                    println!("{:?}", pair);
+                }
+                _ => break,
+            }
+        }
+    }
+
+    fn dels<P>(client: &TxnClient, pairs: P)
```

Comment:
It is not pairs, only keys?

Reply:
Oops. That was a copy paste mistake. Fixed.